s30020-NikiforKoseda/zadanie5.txt

```diff
@@ -0,0 +1,382 @@
+
+Szybkie przemyślenie - czy ustawienie czegoś takiego nie podnosi ryzyka na automatycznie odpalane internetowe skrypty mogące przejąć kotrolę nad komputerem, lub zebrać informacje, jak na memach z losowo odpalającym się okienkiem CMD?
+
+#
+# Zanim zaczniesz czytać i wykonywać dalej instrukcje, które zawarłem poniżej zwróć uwagę na to jak to jest napisane
+# Wszystko to co możesz bez problemu uruchomić ma przed sobą sam znak "#" a po nim NIE MA spacji. Jeśli jest spacja
+# to jest to komentarz tak jak ten tu. W linii 25 masz coś co można uruchomić.
+#
+#
+#
+# Gdybyś potrzebował pomocy z dowolnym poleceniem by zobaczyć jaką ma składnię to pomoc używa się tak:
+#Get-Help Get-ExecutionPolicy
+#
+# A żeby zobaczyć przykłady użycia:
+#Get-Help Get-ExecutionPolicy -Examples
+#
+
+Czyli tak samo jak w wypadku Basha - jeżeli czegoś nie wiemy to udajemy się do manuala. Tylko tutaj nazywa się Get-Help
+
+# Komendy w PS (powershellu) mają pewną stałą konwencję nazewniczą. Jest to: Czasownik-NaCzymMaOperować
+# Np. Get-ExecutionPolicy, Set-ExecutionPolicy, Find-Module, Add-User, Create-Item itd...
+# Nazwa zawsze jest zbudowana w ten sam sposób.
+# Także aby zobaczyć jaki masz "ExecutionPolicy" robisz "Get-ExecutionPolicy" i dostaniesz informację
+#
+# PS ma wiele modułów, które są domyślnie zainstalowane ale nie uruchomione i trzeba je na żądanie ładować.
+# Ale najpierw trzeba wiedzieć co można załadować
+#
+# Poniższa linia zwróci Tobie wszystkie dostepne w danej chwili moduły. Może się chwilę wykonywać bo najpierw musi zebrać listę.
+#Get-Module -ListAvailable
+#
+#
+
+Widzę trzy typy modułów - manifest, script i binary. Po weryfikacji stwierdzam, że:
+Manifest zawiera informacje o autorze modułu i kontroluje niektóre aspekty działania modułu.
+Script to po prostu plik zawierający skrypt z rozszerzeniem psm1 zamiast ps1.
+Binary moduł zawiera klasy i jest zapisany w skompilowanym języku.
+
+#
+# INFO: Poniższa sekcja została przygotowana na Windows 10. Na Win 7 niektóre dane są niedostępne.
+#
+# To co zwróci powyższe polecenie to lista/kolekcja, którą można przetworzyć.
+# Powiedzmy, że chcemy wyszukać tylko te moduły, których wersja jest większa lub równa 2.0 a wszystkie inne nie.
+# Wtedy korzysta się z "Where-Object" lub jego skrótu "?"
+#
+# Znak "|" to potok (pipeline), który uruchamia kolejne polecenie na danych wyjściowych z poprzedniego polecenia
+# Jeśli podaje się kolekcję/listę/tablicę to poszczególne elementy tej kolecji po znaku | są widoczne jako "$_"
+# To taka specjalna zmienna, która jest automatycznie tworzona przez powershell'a kiedy się operuje na kolekcjach.
+#
+# Poniższa linia zwróci tylko te moduły, których wersja jest >= 2.0.0.0
+#Get-Module -ListAvailable | ? {$_.Version -ge "2.0.0.0"}
+#
+
+Czyli pierwsza część polecenia tworzy listę modułów i przekierowuje ją na drugą część, która wyznacza tylko te wiersze, gdzie wersja wynosi powyżej 2.0.0.0. Nie mogę dojść do tego co robi parametr -ge.
```

Comment:
eq - equals
ne - not equals
gt - greather than
ge - greather or equals
lt - less than
le - less or equals

Poza tym to po prostej dedukcji z treści zadania wynika:
> których wersja jest >= 2.0.0.0

a w podanej linicje mamy:
> `$_.Version -ge "2.0.0.0"`

Oczywiste prawda?